subprojects/platform-jvm/src/main/java/org/gradle/jvm/tasks/Jar.java

```diff
@@ -202,4 +218,25 @@ public CopySpec getMetaInf() {
     public CopySpec metaInf(Closure<?> configureClosure) {
         return ConfigureUtil.configure(configureClosure, getMetaInf());
     }
+
+    /**
+     * Provides a comparator that sorts the contents of a Jar/Ear/War so that META-INF is always first.
+     */
+    private static class JarContentsComparator implements Comparator<RelativePath> {
+        @Override
+        public int compare(RelativePath o1, RelativePath o2) {
+            boolean o1MetaInf = isMetaInf(o1.getSegments()[0]);
```

Comment:
I think it is not enough to put the `META-INF` directory first. We need to put the `META-INF/MANIFEST.MF` file first, no matter what else is in the `META-INF` directory.